core/adapters/eth_tx.go

```diff
@@ -224,17 +256,19 @@ func init() {
 
 // Note we need to include the data prefix handling here because
 // if dynamic types (such as bytes) are used, the offset will be affected.
-func getTxDataUsingABIEncoding(e *EthTx, inputData models.JSON) ([]byte, error) {
+func getTxDataUsingABIEncoding(encodingSpec []string, jsonValues []gjson.Result) ([]byte, error) {
 	var arguments abi.Arguments
-	jsonValues := inputData.Get(models.ResultCollectionKey).Array()
-	if len(jsonValues) != len(e.ABIEncoding) {
-		return nil, errors.Errorf("number of collectors %d != number of types in ABI encoding %d", len(jsonValues), len(e.ABIEncoding))
+	if len(jsonValues) != len(encodingSpec) {
+		return nil, errors.Errorf("number of collectors %d != number of types in ABI encoding %d", len(jsonValues), len(encodingSpec))
 	}
 	var values = make([]interface{}, len(jsonValues))
-	for i, argType := range e.ABIEncoding {
+	for i, argType := range encodingSpec {
 		if _, supported := solidityTypeToGoType[argType]; !supported {
 			return nil, errors.Wrapf(ErrInvalidABIEncoding, "%v is unsupported, supported types are %v", argType, supportedSolidityTypes)
 		}
+		if _, ok := jsonTypes[jsonValues[i].Type][argType]; !ok {
```

Comment:
This can't give nil pointer errors if the first element of the map is missing, right?

Reply:
right